### DISCUSSION TITLE
any index for IP inner product ？

### DISCUSSION THREAD

**Discussion by ucas010** [facebookresearch/faiss#4051] [Q&A]
in the [wiki](https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index),
i find the index all for L2 distance , but the datasets(vectors) have been L2 norm , it's not good to use L2 distance ,
so , any index for IP  distance  for 100M ~1B datasets ?
thx

**Comment by mnorris11** [comment]
I don't think I fully understood the question, but you can specify the metric you want to use (including Inner Product) when creating a FAISS index.

**Reply by ucas010** [reply]
In my vectors , processing with L2 norm , then use 
**faiss.index_factory(768, "PCA128,IVF16384_HNSW64,Flat")**
to get the index ,
but the results are the same with different query for recall 
and the distances are as follow,
    "dis": [
        0.2299024760723114,
        0.2320370078086853,
        0.233591690659523,
        0.23407505452632904,
        0.2347213327884674,
        0.23807403445243835,
        0.24065659940242767,
        0.24324321746826172,
        0.24762040376663208,
        0.25289610028266907
    ],

when I set the metric with 
**index =  faiss.index_factory(768, "PCA128,IVF16384_HNSW64,Flat",faiss.METRIC_INNER_PRODUCT)**
the distances 
    "dis": [
        0.23751525580883026,
        0.23581349849700928,
        0.23538577556610107,
        0.23496568202972412,
        0.23209497332572937,
        0.23179949820041656,
        0.23018941283226013,
        0.22953224182128906,
        0.229049414396286,
        0.22857823967933655
    ],

and the results are not similar with query ,either
